Verified under ubuntu 22.04. Compilation fails with
```text
[1923/1947] Building CXX object utils/benchmark/src/CMakeFiles/benchmark.dir/benchmark_register.cc.o
FAILED: utils/benchmark/src/CMakeFiles/benchmark.dir/benchmark_register.cc.o
/usr/bin/c++ -DHAVE_POSIX_REGEX -DHAVE_STD_REGEX -DHAVE_STEADY_CLOCK -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/root/build/utils/benchmark/src -I/root/llvm/utils/benchmark/src -I/root/build/include -I/root/llvm/include -I/root/llvm/utils/benchmark/include -I/root/llvm/utils/benchmark/src/../include -fPIC -fvisibility-inlines-hidden -Werror=date-time -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wno-missing-field-initializers -pedantic -Wno-long-long -Wimplicit-fallthrough -Wno-maybe-uninitialized -Wno-class-memaccess -Wno-redundant-move -Wno-noexcept-type -Wdelete-non-virtual-dtor -Wno-comment -fdiagnostics-color -ffunction-sections -fdata-sections  -std=c++11  -Wall  -Wextra  -Wshadow  -pedantic  -pedantic-errors  -Wfloat-equal  -fstrict-aliasing  -fno-exceptions  -Wstrict-aliasing -O3 -DNDEBUG -std=c++14 -MD -MT utils/benchmark/src/CMakeFiles/benchmark.dir/benchmark_register.cc.o -MF utils/benchmark/src/CMakeFiles/benchmark.dir/benchmark_register.cc.o.d -o utils/benchmark/src/CMakeFiles/benchmark.dir/benchmark_register.cc.o -c /root/llvm/utils/benchmark/src/benchmark_register.cc
In file included from /root/llvm/utils/benchmark/src/benchmark_register.cc:15:
/root/llvm/utils/benchmark/src/benchmark_register.h: In function 'void AddRange(std::vector<T>*, T, T, int)':
/root/llvm/utils/benchmark/src/benchmark_register.h:17:30: error: 'numeric_limits' is not a member of 'std'
   17 |   static const T kmax = std::numeric_limits<T>::max();
      |                              ^~~~~~~~~~~~~~
/root/llvm/utils/benchmark/src/benchmark_register.h:17:46: error: expected primary-expression before '>' token
   17 |   static const T kmax = std::numeric_limits<T>::max();
      |                                              ^
/root/llvm/utils/benchmark/src/benchmark_register.h:17:49: error: '::max' has not been declared; did you mean 'std::max'?
   17 |   static const T kmax = std::numeric_limits<T>::max();
      |                                                 ^~~
      |                                                 std::max
In file included from /usr/include/c++/11/algorithm:62,
                 from /root/llvm/utils/benchmark/include/benchmark/benchmark.h:175,
                 from /root/llvm/utils/benchmark/src/internal_macros.h:4,
                 from /root/llvm/utils/benchmark/src/check.h:8,
                 from /root/llvm/utils/benchmark/src/benchmark_register.h:6,
                 from /root/llvm/utils/benchmark/src/benchmark_register.cc:15:
/usr/include/c++/11/bits/stl_algo.h:3467:5: note: 'std::max' declared here
 3467 |     max(initializer_list<_Tp> __l, _Compare __comp)
      |     ^~~
```

But applying patch to LLVM source tree directly looks somewhat aggressive, IMO.

Replacing defined variables for cmake as
```bash
cmake -G "Ninja" \
      -DCMAKE_BUILD_TYPE=Release \
      -DLLVM_TARGETS_TO_BUILD="X86" \
      -DLLVM_BINUTILS_INCDIR=/usr/include \
      -DLLVM_ENABLE_PROJECTS="clang;compiler-rt" \
      -DLLVM_BUILD_TESTS=OFF \
      -DLLVM_INCLUDE_TESTS=OFF \
      -DLLVM_BUILD_BENCHMARKS=OFF \
      -DLLVM_INCLUDE_BENCHMARKS=OFF \
      ../llvm
```
may be better. It disables the tests and benchmark, and thus fixes the issue.

--- a/build.sh
+++ b/build.sh
@@ -37,7 +37,6 @@ pushd llvm_tools
 wget -O llvm-11.0.0.src.tar.xz https://github.com/llvm/llvm-project/releases/download/llvmorg-11.0.0/llvm-11.0.0.src.tar.xz
 tar -xf llvm-11.0.0.src.tar.xz
 mv      llvm-11.0.0.src        llvm
-sed -i -e 's/#include <vector>/#include <limits>\n#include <vector>/' ./llvm/utils/benchmark/src/benchmark_register.h
 
 wget -O  clang-11.0.0.src.tar.xz https://github.com/llvm/llvm-project/releases/download/llvmorg-11.0.0/clang-11.0.0.src.tar.xz
 tar -xf  clang-11.0.0.src.tar.xz
@@ -56,6 +55,10 @@ cmake -G "Ninja" \
       -DLLVM_TARGETS_TO_BUILD="X86" \
       -DLLVM_BINUTILS_INCDIR=/usr/include \
       -DLLVM_ENABLE_PROJECTS="clang;compiler-rt" \
+      -DLLVM_BUILD_TESTS=OFF \
+      -DLLVM_INCLUDE_TESTS=OFF \
+      -DLLVM_BUILD_BENCHMARKS=OFF \
+      -DLLVM_INCLUDE_BENCHMARKS=OFF \
       ../llvm
 ninja; ninja install
 

